app/src/main/java/com/example/trainingintive/data/repository/UserRepository.kt

```diff
@@ -0,0 +1,8 @@
+package com.example.trainingintive.data.repository
+
+import com.google.firebase.auth.FirebaseAuth
+import javax.inject.Inject
+
+class UserRepository @Inject constructor() {
```

Comment:
I think that you missed domain layer repository interface, you are directly injecting data repo into use case.